lib/go/ecdata/data_source.go

```diff
@@ -0,0 +1,22 @@
+package ecdata
+
+import (
+	"time"
+)
+
+type DataSource interface {
```

Comment:
When I think of a "data source," I tend to think more high-level, like this might be a singleton object for the lifetime of the process.  It looks like it is actually 1:1 tied to a single edge context though, so maybe a name that reflects that?  EdgeContextManipulator or something maybe?